### PR TITLE
added basic xcm configuration

### PR DESCRIPTION
**Pull request by itsmemz**

### CODE REVIEW

File: runtime/arctic/src/lib.rs [hunk 1]
```diff
@@ -491,30 +492,34 @@ impl pallet_ethereum::Config for Runtime {
...
-	type MetadataDepositPerByte = MetaDataDepositPerByte;
-	type ApprovalDeposit = ApprovalDeposit;
-	type StringLimit = StringLimit;
-	type Freezer = ();
-	type Extra = ();
-	type WeightInfo = pallet_assets::weights::SubstrateWeight<Runtime>;
-	type AssetAccountDeposit = ();
+    type Event = Event;
+    type Balance = Balance;
+    type AssetId = AssetId;
```
Comment (mohamedelshami) [inline runtime/arctic/src/lib.rs:511]: Are we sure about the change from u32 to u128?

File: runtime/arctic/src/xcm_config.rs [hunk 2]
```diff
@@ -73,92 +68,152 @@ pub type XcmOriginToTransactDispatchOrigin = (
...
-
-/// Local origins on this chain are allowed to dispatch XCM sends/executions.
+/// No local origins on this chain are allowed to dispatch XCM sends/executions.
 pub type LocalOriginToLocation = SignedToAccountId32<Origin, AccountId, RelayNetwork>;
 
 /// The means for routing XCM messages which are not for local execution into the right message
 /// queues.
 pub type XcmRouter = (
 	// Two routers - use UMP to communicate with the relay chain:
-	cumulus_primitives_utility::ParentAsUmp<ParachainSystem, PolkadotXcm>,
```
Comment (mohamedelshami) [inline runtime/arctic/src/xcm_config.rs:138]: I guess we still need the second router PolkadotXcm?

File: runtime/arctic/src/xcm_config.rs [hunk 3]
```diff
@@ -1,35 +1,30 @@
...
-	Runtime, XcmpQueue,
+	AccountId, Balances, Call, Event, Origin, ParachainInfo, ParachainSystem, PolkadotXcm, Runtime,
+	WeightToFee, XcmpQueue,
 };
+use core::marker::PhantomData;
 use frame_support::{
-	match_type, parameter_types,
+	log, match_types, parameter_types,
 	traits::{Everything, Nothing},
-	weights::{IdentityFee, Weight},
```
Comment (mohamedelshami) [inline runtime/arctic/src/xcm_config.rs:8]: Consider using WeightToFee here, or how do we configure the fee here as opposed to the transaction payment fee?

File: runtime/arctic/src/lib.rs [hunk 4]
```diff
@@ -435,6 +434,8 @@ parameter_types! {
 	pub MinimumMultiplier: Multiplier = Multiplier::saturating_from_rational(1, 1_000_000_000u128);
 }
 
+pub type WeightToFee = constants::fee::WeightToFee;
```
Comment (mohamedelshami) [inline runtime/arctic/src/lib.rs:437]: Any reason we can't assign this directly rather than wrapping it up in another type?

**Review by mohamedelshami — CHANGES_REQUESTED** [review]

File: runtime/arctic/src/lib.rs (hunk 1 above)
Comment (itsmemz) [reply runtime/arctic/src/lib.rs:511]: I am not sure why we added u32, but it is a recommended assetId type in vanilla template, as there can be long assets id also in the mainnet, so I think its fine to have like this

File: runtime/arctic/src/xcm_config.rs (hunk 2 above)
Comment (itsmemz) [reply runtime/arctic/src/xcm_config.rs:138]: Second router is still there, just second parameter was removed, which was xcm version currently supported by network, its an optional parameter though, but I have added back

**Pull request merged**